src/routes/safe/store/actions/transactions/utils/transactionHelpers.ts

```diff
@@ -26,18 +26,35 @@ import { TRANSACTIONS_REDUCER_ID } from 'src/routes/safe/store/reducer/transacti
 import { store } from 'src/store'
 import { safeSelector, safeTransactionsSelector } from 'src/routes/safe/store/selectors'
 import { addOrUpdateTransactions } from 'src/routes/safe/store/actions/transactions/addOrUpdateTransactions'
-import { TxServiceModel } from 'src/routes/safe/store/actions/transactions/fetchTransactions/loadOutgoingTransactions'
+import {
+  BatchProcessTxsProps,
+  TxServiceModel,
+} from 'src/routes/safe/store/actions/transactions/fetchTransactions/loadOutgoingTransactions'
 import { TypedDataUtils } from 'eth-sig-util'
+import { Token } from 'src/logic/tokens/store/model/token'
+import { SafeRecord } from 'src/routes/safe/store/models/safe'
 
 export const isEmptyData = (data?: string | null): boolean => {
   return !data || data === EMPTY_DATA
 }
 
-export const isInnerTransaction = (tx: TxServiceModel, safeAddress: string): boolean => {
-  return sameAddress(tx.to, safeAddress) && Number(tx.value) === 0
+export function isInnerTransaction(tx: Transaction, safeAddress: string): boolean
+export function isInnerTransaction(tx: TxServiceModel, safeAddress: string): boolean
+export function isInnerTransaction(tx: any, safeAddress: string): boolean {
```

Comment:
What's going on there?

Reply:
Related to the function overload, the thing is that `isCancelTransaciton` is being used by the tx returned by the transaction service (`tx.to`)... and the one we have stored in the store (`tx.recipient`)